src/client/src/orangehrmRecruitmentPlugin/components/VacancyCard.vue

```diff
@@ -0,0 +1,111 @@
+<!--
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-paper-container">
+    <div class="orangehrm-vacancy-card-header">
+      <oxd-text type="card-title">
+        {{ vacancyTitle }}
+      </oxd-text>
+      <oxd-button
+        :label="$t('general.apply')"
+        class="oxd-button"
+        display-type="success"
+        @click="apply"
+      ></oxd-button>
+    </div>
+    <oxd-divider v-show="vacancyDescription"></oxd-divider>
+    <div :class="{'orangehrm-vacancy-card-body': isViewDetails}">
+      <oxd-text type="toast-message">
+        <pre v-if="vacancyDescription" class="orangehrm-vacancy-card-pre-tag">
+          {{ vacancyDescription }}
+        </pre>
+      </oxd-text>
+    </div>
+    <div
+      v-if="vacancyDescription && vacancyDescription.length > descriptionLength"
+      class="orangehrm-vacancy-card-footer"
+    >
+      <a @click="viewDetails">
+        <oxd-text class="orangehrm-vacancy-card-anchor-tag" type="subtitle-2">
+          {{
+            isViewDetails
+              ? $t('recruitment.show_more')
+              : $t('recruitment.show_less')
+          }}
+        </oxd-text>
+      </a>
+    </div>
+  </div>
+</template>
+
+<script>
+import useResponsive from '@ohrm/oxd/composables/useResponsive';
+import {toRefs} from 'vue';
+import {navigate} from '@/core/util/helper/navigation';
+
+export default {
+  name: 'VacancyCard',
+  props: {
+    vacancyId: {
+      type: Number,
+      required: true,
+    },
+    vacancyTitle: {
+      type: String,
+      required: true,
+    },
+    vacancyDescription: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const responsiveState = useResponsive();
+    return {
+      ...toRefs(responsiveState),
+    };
+  },
+  data() {
+    return {
+      viewMore: false,
+    };
+  },
+  computed: {
+    isViewDetails() {
+      return !this.viewMore;
+    },
+    descriptionLength() {
+      if (this.isMobile) return 150;
+      return this.windowWidth < 1920 ? 250 : 400;
```

Comment:
@Super-Chama can you check this. based on the locale, may have to check.

WWWWWWWWWW
wwwwwwwwww
just check with 10 chars

Reply:
@RajithaKumara , We talked about it.. Agreed for now.. May have to change later.

Reply:
@OrangeHRM-SaifZaman we talked about it. Remember you said you will check the possibility. no agreements were made. check if below solutions works for you. 

https://codepen.io/srekoble/pen/WGBzea?editors=1100
https://css-tricks.com/multiline-truncated-text-with-show-more-button/

Reply:
Not solved